src/libs/meta/meta.c

```diff
@@ -1211,7 +1211,7 @@ int elektraSortTopology (KeySet * ks, Key ** array)
 					retVal = -1;
 					break;
 				}
-				else if (i == j)
+				else if ((long)i == (long)j)
 				{
```

Comment:
so, if a dependency is a valid keyName, but doesn't exist it should be ignored ? 
that'd be no problem.